prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
     return 1
 }
 
-createTestResources() {
-    shout "Create e2e upgrade test resources"
-    date
+function installTestChartOrFail() {
+  local path=$1
+  local name=$2
+  local namespace=$3
 
-    injectTestingAddons
+  shout "Create ${name} resources"
+  date
 
-    if [  -f "$(helm home)/ca.pem" ]; then
-        local HELM_ARGS="--tls"
-    fi
+  local HELM_ARGS
+  if [[ -f "$(helm home)/ca.pem" ]]; then
+      HELM_ARGS="--tls"
+  fi
+
+  helm install "${path}" \
+      --name "${name}" \
+      --namespace "${namespace}" \
+      --timeout "${HELM_TIMEOUT_SEC}" \
+      --set domain="${DOMAIN}" \
+      --wait ${HELM_ARGS}
+
+  prepareResult=$?
+  if [[ "${prepareResult}" != 0 ]]; then
+      echo "Helm install ${name} operation failed: ${prepareResult}"
+      exit "${prepareResult}"
+  fi
+}
 
-    helm install "${UPGRADE_TEST_PATH}" \
-        --name "${UPGRADE_TEST_RELEASE_NAME}" \
-        --namespace "${UPGRADE_TEST_NAMESPACE}" \
-        --timeout "${UPGRADE_TEST_HELM_TIMEOUT_SEC}" \
-        --wait ${HELM_ARGS}
+function waitForTestPodToFinish() {
+  local name=$1
+  local namespace=$2
+  local label=$3
+
+  set +o errexit
+  checkTestPodTerminated "${namespace}"
+  prepareTestResult=$?
+  set -o errexit
+
+  echo "Logs for prepare data operation to ${name}: "
+  # shellcheck disable=SC2046
+  kubectl logs -n "${namespace}" $(kubectl get pod -n "${name}" -l "${label}=${TEST_RESOURCE_LABEL_VALUE_PREPARE}" -o json | jq -r '.items | .[] | .metadata.name') -c "${TEST_CONTAINER_NAME}"
+  if [[ "${prepareTestResult}" != 0 ]]; then
+      echo "Exit status for prepare ${name}: ${prepareTestResult}"
+      exit "${prepareTestResult}"
+  fi
+}
 
-    prepareResult=$?
-    if [ "${prepareResult}" != 0 ]; then
-        echo "Helm install operation failed: ${prepareResult}"
-        exit "${prepareResult}"
-    fi
+createTestResources() {
+    injectTestingAddons
 
-    set +o errexit
-    checkTestPodTerminated
-    prepareTestResult=$?
-    set -o errexit
-
-    echo "Logs for prepare data operation to test e2e upgrade: "
-    # shellcheck disable=SC2046
-    kubectl logs -n "${UPGRADE_TEST_NAMESPACE}" $(kubectl get pod -n "${UPGRADE_TEST_NAMESPACE}" -l "${UPGRADE_TEST_RESOURCE_LABEL}=${UPGRADE_TEST_LABEL_VALUE_PREPARE}" -o json | jq -r '.items | .[] | .metadata.name') -c "${TEST_CONTAINER_NAME}"
-    if [ "${prepareTestResult}" != 0 ]; then
-        echo "Exit status for prepare upgrade e2e tests: ${prepareTestResult}"
-        exit "${prepareTestResult}"
-    fi
+    installTestChartOrFail "${UPGRADE_TEST_PATH}" "${UPGRADE_TEST_RELEASE_NAME}" "${UPGRADE_TEST_NAMESPACE}"
```

Comment:
Maybe that comment makes it a bit easier to read 😎 
```suggestion
    # install upgrade test
    installTestChartOrFail "${UPGRADE_TEST_PATH}" "${UPGRADE_TEST_RELEASE_NAME}" "${UPGRADE_TEST_NAMESPACE}"
```

Reply:
Done